core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -178,24 +178,23 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     const auto value_count =
```

Comment:
Because we are going to call `std::max` on this, we should name the type (because if the type isn't `unsigned int` `std::max` will fail to compile) instead of using `auto`.